implementation/rest/src/main/java/io/smallrye/opentelemetry/implementation/rest/OpenTelemetryServerFilter.java

```diff
@@ -141,8 +150,7 @@ protected InetSocketAddress getPeerSocketAddress(final ContainerRequestContext r
 
         @Override
         protected InetSocketAddress getHostSocketAddress(final ContainerRequestContext request) {
-            URI requestUri = request.getUriInfo().getRequestUri();
-            return new InetSocketAddress(requestUri.getHost(), requestUri.getPort());
+            return new InetSocketAddress(getHostName(request), getHostPort(request));
```

Comment:
Should we guard against -1 being returned or are we okay with the `InetSocketAddress` throwing the exception?

Reply:
In the case the port can't be found from the scheme then I'd rather have a -1